custom_components/powercalc/sensors/group.py

```diff
@@ -575,24 +578,27 @@ def _async_hide_members(self, hide: bool) -> None:
             registry.async_update_entity(entity_id, hidden_by=hidden_by)
 
     @callback
-    def on_state_change(self, _: Any) -> None:  # noqa
+    def on_state_change(self, event: Event[EventStateChangedData]) -> None:
         """Triggered when one of the group entities changes state."""
 
+        new_state = self.calculate_new_state(event.data.get("new_state"))
```

Comment:
```suggestion
        new_state = self.calculate_new_state(event.data["new_state"])
```

`EventStateChangedData` type shows this as always present

Reply:
Thanks for the suggestion. Is retrieving the value directly like this faster than `get()` on dictionary. Good to know, I'll consider other cases like this when developing in the future.